R/oauth-flow-password.R

```diff
@@ -69,9 +69,20 @@ oauth_flow_password <- function(client,
 
 check_password <- function(password, call = caller_env()) {
   if (is.null(password)) {
-    check_installed("askpass", call = call)
-    password <- askpass::askpass()
+    password <- ask_user_prompt()
   }
   check_string(password, call = call)
   password
 }
+
+ask_user_prompt <- function(prompt = "Please enter your password: ") {
+  if (is_rstudio_session()) {
+    check_installed("rstudioapi")
+    result <- rstudioapi::askForPassword(prompt)
+  } else {
```

Comment:
Should this have an `!is_interactive()` branch with an informative error?

Reply:
I don't _think_ it's needed?
* Within RStudio: `rstudioapi::askForPassword()` still works (surprising, but very welcome) when knitting
![image](https://github.com/r-lib/httr2/assets/56164984/0e3f35d0-7c12-42b3-8261-dfe1fe1dd70e)
* If running outside of RStudio (e.g., Connect) then `is_rstudio_session()` is already false

Are there other cases I might be overlooking?

Reply:
There are two scenarios running outside of RStudio that I'm worried about:

* You're running R interactively and call `knit()`: does `readline()` cause execution to pause? Do you see the correct prompt?
* You're running R in batch mode and call `knit()`: won't `readline()` will return `""` causing an authentication problem that's hard to diagnose?

Reply:
Any more thoughts on this comment? I think it's the main blocker for merging this PR.

Reply:
Sorry for the delay while juggling some other things. Of note, this PR is a specific workaround for Snowflake auth in Workbench and was before I chatted with @atheriel about his efforts to integrate OAuth more directly via Workbench (rather than our current wrappers around `httr2` which have been reasonably well).

Regardless, I had a chance to think a bit more about this, motivated by [Quarto's execution approach not supporting interactive user prompts](https://github.com/quarto-dev/quarto-cli/discussions/6076).

I've pushed a branch, per your suggestion and very helpful test cases! I _think_ it's closer to a reasonable approach.

Current behavior:

1. In RStudio: 
    a. .rmd -> knitting: pauses to receive prompt (new, yay!)
    b. .qmd -> render: fails with an expected, or at least, [well-documented](https://github.com/quarto-dev/quarto-cli/discussions/6076) error:
```
Error:
! RStudio not running
```
2. Outside of RStudio: both of your test cases:
```
## Error in `prompt_user()` at script.R:18:3:
## ! Unable to obtain user input in a non-interactive session.
```

Open to any thoughts!